src/api.cc

```diff
@@ -214,8 +175,43 @@ CheckInResult AkliteClient::CheckIn() const {
   pt.put<std::string>("target", current.filename());
   pt.put<std::string>("ostreehash", current.sha256Hash());
 
+  LOG_INFO << "Updating the local TUF repo with metadata located in " << client_->config.uptane.repo_server << "...";
   auto repo_src = std::make_shared<aklite::tuf::AkHttpsRepoSource>("temp-remote-repo-source", pt, client_->config);
-  return UpdateMetaAndGetTargets(repo_src);
+  CheckInResult::Status check_status{CheckInResult::Status::Failed};
+  try {
+    tuf_repo_->updateMeta(repo_src);
+    check_status = CheckInResult::Status::Ok;
+    LOG_INFO << "The local TUF repo has been successfully updated";
+  } catch (const std::exception& exc) {
+    const std::string err{exc.what()};
+    LOG_ERROR << "Failed to update the local TUF repo: " << err;
+    if (err.find("Failed to fetch role timestamp") != std::string::npos) {
+      LOG_ERROR << "Check the device tag or verify the existence of a wave for the device";
+    }
+  }
+  if (check_status != CheckInResult::Status::Ok) {
+    LOG_INFO << "Checking the local TUF repo...";
+    try {
+      tuf_repo_->checkMeta();
+      check_status = CheckInResult::Status::OkCached;
+      LOG_INFO << "The local TUF is valid";
+    } catch (const std::exception& exc) {
+      LOG_ERROR << "Local TUF repo is invalid: " << exc.what();
+    }
+  }
+
+  if (check_status == CheckInResult::Status::Failed) {
+    return CheckInResult{check_status, "", {}};
+  }
+
+  LOG_INFO << "Searching for matching TUF Targets...";
+  auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, client_->tags, secondary_hwids_);
+  if (matchingTargets.empty()) {
+    LOG_ERROR << "No matching Targets are found for a given device; check a device tag or a hardware ID";
```

Comment:
```suggestion
    LOG_ERROR << "There are no Targets matching the current device tag and hardware ID";
```

Reply:
Can we add the verb `found` or synonym to this error message so it goes inline with the `"Searching for matching TUF Targets..." message. 
Maybe something like "Couldn't find..." or "Failed to find"?

Reply:
Sure, makes sense. Adjusted the suggestions accordingly.

Reply:
updated via force push